plugin/trino-mysql/src/main/java/io/trino/plugin/mysql/MySqlClient.java

```diff
@@ -471,6 +486,17 @@ public Optional<ColumnMapping> toColumnMapping(ConnectorSession session, Connect
         return Optional.empty();
     }
 
+    private static LongReadFunction timestampMillisReadFunction()
+    {
+        return (resultSet, columnIndex) -> toTrinoTimestampMillis(resultSet.getObject(columnIndex, LocalDateTime.class));
+    }
+
+    private static long toTrinoTimestampMillis(LocalDateTime localDateTime)
+    {
+        return localDateTime.toEpochSecond(UTC) * MICROSECONDS_PER_SECOND
+                + Math.round(localDateTime.getNano() * 1.0 / NANOSECONDS_PER_MILLISECOND) * MICROSECONDS_PER_MILLISECOND;
```

Comment:
For timestamp precision rounding see
https://github.com/trinodb/trino/blob/e71e0b5b6d2b715f27d7fc0650386684c6572a31/plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/StandardColumnMappings.java#L543